osu.Game.Tests/Chat/MessageFormatterTests.cs

```diff
@@ -428,23 +428,23 @@ public void TestLinkComplex()
             Assert.AreEqual(5, result.Links.Count);
 
             Link f = result.Links.Find(l => l.Url == "https://osu.ppy.sh/wiki/wiki links");
-            Assert.AreEqual(44, f.Index);
+            Assert.AreEqual(44, f!.Index);
             Assert.AreEqual(10, f.Length);
 
             f = result.Links.Find(l => l.Url == "http://www.simple-test.com");
-            Assert.AreEqual(10, f.Index);
+            Assert.AreEqual(10, f!.Index);
```

Comment:
[sharplab](https://sharplab.io/#v2:D4YgdgrgNlCGBGUCmACJYHILAChcAEAmARgIGYUiUBhXAb1xScovwBYUBZACn2IAYA/CgDOASkbMGOZrJQBLMABcFKALyiAhADoAMugDmSgBYBuSXNUBqDSL2GT5mcwC+uF0A===)

The compiler and the language say that, a first `!` will imply the variable being non-null in the following.

Reply:
The f is not the same.